imap_processing/mag/l1a/mag_l1a_data.py

```diff
@@ -92,30 +157,87 @@ class MagL1a:
+        Dictionary of packet properties for each packet in the day. The key is the start
+        time of the packet, and the value is a dataclass of packet properties.
+    most_recent_sequence : int
+        Sequence number of the most recent packet added to the object
+    missing_sequences : list[int]
+        List of missing sequence numbers in the day
+    start_time : numpy.datetime64
+        Start time of the day, in ns since J2000 epoch
     """
 
     is_mago: bool
-    active: bool
-    SHCOARSE: int
-    vectors: np.ndarray
+    is_active: int
+    shcoarse: int
+    vectors: np.array
+    starting_packet: InitVar[MagL1aPacketProperties]
+    packet_definitions: dict[np.datetime64, MagL1aPacketProperties] = field(init=False)
+    most_recent_sequence: int = field(init=False)
+    missing_sequences: list[int] = field(default_factory=list)
+    start_time: np.datetime64 = field(init=False)
+
+    def __post_init__(self, starting_packet: MagL1aPacketProperties) -> None:
+        """
+        Initialize the packet_definition dictionary and most_recent_sequence.
+
+        Parameters
+        ----------
+        starting_packet : MagL1aPacketProperties
+            The packet properties for the first packet in the day, including start time.
+        """
+        # TODO should this be from starting_packet
+        self.start_time = (J2000_EPOCH + met_to_j2000ns(self.shcoarse)).astype(
+            "datetime64[D]"
+        )
+        self.packet_definitions = {self.start_time: starting_packet}
+        # most_recent_sequence is the sequence number of the packet used to initialize
+        # the object
+        self.most_recent_sequence = starting_packet.src_seq_ctr
+
+    def append_vectors(
+        self, additional_vectors: np.array, packet_properties: MagL1aPacketProperties
+    ) -> None:
+        """
+        Append additional vectors to the current vectors array.
+
+        Parameters
+        ----------
+        additional_vectors : numpy.array
+            New vectors to append.
+        packet_properties : MagL1aPacketProperties
+            Additional vector definition to add to the l0_packets dictionary.
+        """
+        vector_sequence = packet_properties.src_seq_ctr
+
+        self.vectors = np.concatenate([self.vectors, additional_vectors])
+        self.packet_definitions[self.start_time] = packet_properties
+
+        # Every additional packet should be the next one in the sequence, if not, add
+        # the missing sequence(s) to the gap data
+        if not self.most_recent_sequence + 1 == vector_sequence:
+            self.missing_sequences += list(
+                range(self.most_recent_sequence + 1, vector_sequence)
+            )
+        self.most_recent_sequence = vector_sequence
 
     @staticmethod
     def calculate_vector_time(
-        vectors: np.ndarray, vectors_per_second: int, start_time: TimeTuple
-    ) -> np.ndarray:
+        vectors: np.ndarray, vecters_per_sec: int, start_time: TimeTuple
```

Comment:
```suggestion
        vectors: np.ndarray, vectors_per_sec: int, start_time: TimeTuple
```

Reply:
Oh wow good catch! Spell checker missed it